File: java/smartlist/src/main/java/com/smartlist/module/grading/dto/request/QuestionScoreReqDTO.java
package com.smartlist.module.grading.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class QuestionScoreReqDTO {

    @Schema(description = "Assessment Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value="assessment_id")
    Integer assessmentId;

    @Schema(description = "Question Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value="question_id")
    Integer questionId;

    @Schema(description = "Score", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer score;


}

File: javascript/smartlist/smartlist-api-mongodb-project/src/modules/grading/dto/request/question-score.req.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty } from 'class-validator';
export class QuestionScoreReqDto {
  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  assessment_id: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  question_id: string;

  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  score: number;
}
